api/handlers/v1/server.go

```diff
@@ -178,6 +182,51 @@ func (server APIServer) ApplyAction(ctx context.Context, request *entropyv1beta1
 	return response, nil
 }
 
+func (server APIServer) CreateProvider(ctx context.Context, request *entropyv1beta1.CreateProviderRequest) (*entropyv1beta1.CreateProviderResponse, error) {
+	pro := providerFromProto(request.Provider)
+	pro.Urn = domain.GenerateProviderUrn(pro)
+	// TODO: add provider validation
+
+	createdProvider, err := server.providerService.CreateProvider(ctx, pro)
+	if err != nil {
+		if errors.Is(err, store.ErrProviderAlreadyExists) {
+			return nil, status.Error(codes.AlreadyExists, "provider already exists")
+		}
+		return nil, ErrInternal
+	}
+
+	responseProvider, err := providerToProto(createdProvider)
+	if err != nil {
+		return nil, ErrInternal
+	}
+	response := entropyv1beta1.CreateProviderResponse{
+		Provider: responseProvider,
+	}
+	return &response, nil
+}
+
+func (server APIServer) ListProviders(ctx context.Context, request *entropyv1beta1.ListProvidersRequest) (*entropyv1beta1.ListProvidersResponse, error) {
+	var responseProviders []*entropyv1beta1.Provider
+	providers, err := server.providerService.ListProviders(ctx, request.GetParent(), request.GetKind())
+	if err != nil {
+		return nil, ErrInternal
+	}
+	for _, pro := range providers {
+		responseProvider, err := providerToProto(pro)
+		if err != nil {
+			return nil, ErrInternal
+		}
+		responseProviders = append(responseProviders, responseProvider)
+	}
+	if err != nil {
```

Comment:
This extra block of error check is unnecessary.

Reply:
Oh, my bad! Fixed this.